### PR TITLE
Compiler error for signature/implementation files when declaring multiple parameters with the same name in abstract methods

### PR DESCRIPTION
**Pull request by edgarfgp**
- Supersedes https://github.com/dotnet/fsharp/pull/13964
-  Fixes https://github.com/dotnet/fsharp/issues/13148

### CODE REVIEW

**Review by T-Gro — APPROVED** [review]

**Review by 0101 — DISMISSED** [review]

File: tests/FSharp.Compiler.ComponentTests/Conformance/DeclarationElements/MemberDefinitions/MethodsAndProperties/MethodsAndProperties.fs [hunk 1]
```diff
@@ -414,4 +414,163 @@ module MethodsAndProperties =
...
+            |> withDiagnostics [
+            (Error 3549, Line 5, Col 5, Line 5, Col 38, "Duplicate parameter. The parameter 'i' has been used more that once in this method.")
+            (Error 3549, Line 7, Col 5, Line 7, Col 39, "Duplicate parameter. The parameter 'i' has been used more that once in this method.")
+            (Error 3549, Line 9, Col 5, Line 9, Col 47, "Duplicate parameter. The parameter 'i' has been used more that once in this method.")
+            (Error 3549, Line 11, Col 5, Line 11, Col 54, "Duplicate parameter. The parameter 'i' has been used more that once in this method.")
+            (Error 3549, Line 11, Col 5, Line 11, Col 54, "Duplicate parameter. The parameter 'j' has been used more that once in this method.")
+        ]
+        
+    [<Fact>]
+    let ``Error in signature file with not implementation file with abstract methods when reusing parameters`` () =
```
Comment (psfinaki) [inline tests/FSharp.Compiler.ComponentTests/Conformance/DeclarationElements/MemberDefinitions/MethodsAndProperties/MethodsAndProperties.fs:442]:
```suggestion
    let ``Error in signature file with no implementation file with abstract methods when reusing parameters`` () =
```

**Review by psfinaki — APPROVED** [review]
Thanks, good fix.

**Comment by psfinaki** [comment]
@T-Gro @0101 I fixed the conflict and a typo - please reapprove.

**Review by 0101 — APPROVED** [review]

**Comment by edgarfgp** [comment]
@psfinaki it seems like the CI does not like the changes on FSComp.txt . Had the the same issue in the past and I had to create a brand new beach and start from scratch 😔

**Comment by psfinaki** [comment]
@edgarfgp damn. It's probably worth taking a closer look on why is this happening, will come back to this tomorrow.

**Review by psfinaki — APPROVED** [review]

**Review by 0101 — APPROVED** [review]

**Comment by psfinaki** [comment]
@edgarfgp alright, there it is, thanks again :)

**Pull request merged**